DatadogCrashReporting/Sources/KSCrashIntegration/KSCrashBacktrace.swift

```diff
@@ -112,3 +114,37 @@ internal struct KSCrashBacktrace: BacktraceReporting {
         return String(cString: buffer)
     }
 }
+
+/// Symbolicate an address and return full binary image information including CPU type.
+///
+/// This function bypasses KSCrash's `ksbt_symbolicateAddress()` because that function only
+/// returns a subset of binary image information via `KSSymbolInformation`, which lacks the
+/// CPU type and subtype fields. Instead, we directly call the underlying `ksdl_*` functions
+/// to get the complete `KSBinaryImage` struct, which includes `cpuType` and `cpuSubType`.
+/// This allows us to determine the correct architecture (arm64, arm64e, etc.) for each
+/// binary image, which is critical for accurate stacktrace symbolication.
+private func symbolicate(address: uintptr_t) -> KSBinaryImage? {
```

Comment:
Nice! Please push a PR to KSCrash so we can incorporate this into the main repo.